src/components/Item/Item.js

```diff
@@ -146,8 +149,13 @@ export const Item = ({
 
   const avatar = <img {...{ src: items[id] }} alt={name} />
 
-  const sellPrice =
-    adjustedValue * getSalePriceMultiplier(completedAchievements)
+  let sellPrice = adjustedValue
+
+  // #140 - never increase the value of items the shop sells otherwise they
+  // can be bought and instantly resold for a profit to game the.. game
+  if (!shopItemIds.includes(id)) {
+    sellPrice *= getSalePriceMultiplier(completedAchievements)
+  }
```

Comment:
This logic all looks sound to me! I really like the simplicity of this solution.

I'm a bit concerned about the performance cost of doing the `O(n)` lookup operation (`shopItemIds.includes(id)`), as this is a fairly hot (or at least warm) code path. While `shopItemIds` isn't a massive data set, I think it should be pretty easy to change this to be an `O(1)` operation by changing `shopItemIds` to being a `Set`. So on line 36, we could set it up to be:

```js
const shopItemIds = new Set(shopInventory.map(item => item.id))
```

And on line 156, the check would then be:

```js
if (!shopItemIds.has(id))
```

(For reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#performance)

We could do this with another data structure as well (like a plain object). What do you think @lstebner?

Reply:
a very valid concern. glad you brought it up. i like the idea of using a Set a lot, i was also thinking this could maybe be exported from the shop inventory instead of calculated in the component. how does that sound to you?

Reply:
Sounds smart to me, let’s do it! 👍

Reply:
updated! @jeremyckahn